KoPrivateGPT/retrieval/base.py

```diff
@@ -18,3 +25,70 @@ def ingest(self, passages: List[Passage]):
     @abstractmethod
     def retrieve_id(self, query: str, top_k: int = 5, *args, **kwargs) -> List[Union[str, UUID]]:
         pass
+
+    def fetch_data(self, ids: List[UUID]) -> List[Passage]:
+        db_origin_list = self.redis_db.get_json(ids)
+        # Sometimes redis doesn't find the id, so we need to filter that db_origin is None.
+        filter_db_origin = list(filter(lambda db_origin: db_origin is not None, db_origin_list))
+        # Check duplicated db instance
+        final_db_origin = self.duplicate_check(filter_db_origin)
+        # fetch data from each db
+        passage_list = self.fetch_each_db(final_db_origin, ids)
+        return passage_list
+
+    def fetch_each_db(self, final_db_origin: dict[tuple, list[int]], ids: List[UUID]):
+        """
+        check_dict = {(("db_type": "mongo_db"),
+            (('mongo_url': "~"), ('db_name': "~"), ('collection_name': "~"))): [0,  2], ...}
+        """
+        fetch_list = []
+        for item in final_db_origin.items():
+            # make tuple to dict
+            # item[0] = (db_origin:tuple), item[1] = (index:list)
+            db_origin = dict(item[0])
+            dict_db_path = dict(db_origin['db_path'])
+            # make db instance
+            self.create_db_instance(db_origin['db_type'], dict_db_path)
+            self.db.load()
+            # make each id list
+            each_ids = [ids[i] for i in item[1]]
+            # fetch data
+            fetch_data = (self.db.fetch(each_ids))
+            fetch_list.append(fetch_data)
+        # make flatten list(passage_list) from fetch_list
+        passage_list = list(itertools.chain.from_iterable(fetch_list))
+        return passage_list
+
+    def create_db_instance(self, db_type: str, db_path: dict):
+        """
+        selector-ModuleSelector cant import because of circular import.
+        """
+        if db_type == "mongo_db":
+            self.db = MongoDB(**db_path)
```

Comment:
I don't know why you are assign new BaseDB instance to self.db. Actually, why self.db exists?